stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDOMProvider.java

```diff
@@ -48,6 +50,24 @@
   private final InspectModeHandler mInspectModeHandler;
   private Listener mListener;
 
+  // We don't yet have an an implementation for reliably detecting fine-grained changes in the
+  // View tree. So, for now at least, we have a timer that runs every so often and just reports
+  // that we changed. Our listener will then read the entire DOM from us and transmit the changes to
+  // Chrome. Detecting, reporting, and traversing fine-grained changes is a future work item.
+  private boolean mIsReportChangesTimerPosted = false;
+  private final Runnable mReportChangesTimer = new Runnable() {
+    @Override
+    public void run() {
+      mIsReportChangesTimerPosted = false;
+
+      if (mListener != null) {
+        mListener.onChanged();
```

Comment:
Yes, I meant does this functionally cause the diffing.  I agree it's an implementation detail, but I disagree that it's _exclusively_ an implementation detail.  `onPossiblyChanged` suggests that callers will not be perfectly intelligent, and they never will be because if they could be so intelligent we wouldn't have used this design in the first place.  The idea that we have a master trigger that will rescan and diff the entire DOM suggests we are not smart about changes.  That's part of the API as much as it is part of the implementation detail.  Might as well make that clearer through the API.

Invalidate semantics also work, but then you should actually call it invalidate.  Saying "onChange" tells any reasonable reader that a change has been detected.  The containing code, however, clearly suggests otherwise :)